hbase-server/src/test/java/org/apache/hadoop/hbase/util/TestRegionMover.java

```diff
@@ -457,4 +463,37 @@ public void testExcludeAndDecomServers() throws Exception {
       Collections.emptyList());
   }
 
+  @Test
+  public void testWithMergedRegions() throws Exception {
```

Comment:
test with split regions as well?

test that we properly fail in ack mode if the problem is something other than URE?